support/rmt

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#############################################################
+# Name:        Supportconfig Plugin for RMT
+# Description: Gathers important troubleshooting information
+#              about RMT
+# License:     GPLv2
+# Author:      SCC Team <happy-customer@suse.de>
+# Modified:    2018 April 18
+#############################################################
+
+SVER=0.0.1
+RCFILE="/usr/lib/supportconfig/resources/scplugin.rc"
+
+[ -s $RCFILE ] && . $RCFILE || { echo "ERROR: Initializing resource file: $RCFILE"; exit 1; }
+
+## Our own helper functions
+validate_rpm_if_installed() {
+  THISRPM=$1
+  echo "#==[ Validating RPM ]=================================#"
+  if rpm -q "$THISRPM" >/dev/null 2>&1; then
+    echo "# rpm -V $THISRPM"
+
+    if rpm -V "$THISRPM"; then
+      echo "Status: Passed"
+    else
+      echo "Status: WARNING"
+    fi
+  else
+    echo "package $THISRPM is not installed"
+    echo "Status: Skipped"
+  fi
+  echo
+}
+
+RPMLIST="
+rmt-server
+nginx
+mariadb
```

Comment:
I would suggest adding `openssl`, `libssl` and `libcurl` -- sometimes the customers mess with those and that causes various obscure issues.

Hmm, and maybe getting vendor information somehow would be good too, in case the package is installed from 3rd-party repos.

Reply:
Maybe `rpm -q --whatprovides ssl`?

Reply:
@ikapelyukhin rpm version and vendor information for all rpms is already included by the default supportconfig queries.
```#==[ Command ]======================================#
# rpm -qa --queryformat "%-35{NAME} %-35{DISTRIBUTION} %{VERSION}-%{RELEASE}\n" | sort -k 1,2 -t " " -i
NAME                                DISTRIBUTION                        VERSION
...
libopenssl1_0_0                     SUSE Linux Enterprise 15            1.0.2n-1.19
libopenssl1_0_0-32bit               SUSE Linux Enterprise 12            1.0.2j-60.20.2
libopenssl1_1                       SUSE Linux Enterprise 15            1.1.0g-1.5
libopenssl1_1-32bit                 SUSE Linux Enterprise 15            1.1.0g-1.5
```
as well as the installed products and their services and repositories.

Reply:
The real danger would be people dumping unpackaged openssl binaries somewhere, but supportconfig should already catch those by dumping the output of /usr/bin/env to env.txt.

Reply:
The real danger would be people building libcurl/libssl from source without touching the libs installed from RPMs and messing with `LD_LIBRARY_PATH` :grimacing:
But that's beyond the point. Verifying files in the RPMs and having vendor info should be good enough for most cases :+1:

Reply:
You wouldn't believe it but one of our tricky SUSEConnect bugs was due to a Novell groupwise server installation script futzing with the system openssl bugs in precisely that way. The support engineer didn't look for it as craziness in groupwise packages was 'beyond the point'.